cdap-data-fabric/src/main/java/co/cask/cdap/data2/nosql/NoSqlStructuredTableRegistry.java

```diff
@@ -154,6 +161,21 @@ public boolean isEmpty() {
     );
   }
 
+  @Nullable
+  private StructuredTableSpecification getSpecificationFromStorage(StructuredTableId tableId) {
```

Comment:
Better to have this method return `Optional` instead of exposing `null` here and then wrap it in the cache.

Reply:
Done